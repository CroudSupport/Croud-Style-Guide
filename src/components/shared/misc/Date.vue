<template>
    <span>{{ formattedDate }}</span>
</template>

<script>
    import moment from 'moment'

    /**
     * Preformatted date component
     *
     * @example ./croud-date.md
     */
    export default {
        name: 'croud-date',

        props: {
            /**
             * Moment date object or iso compliant string
             */
            date: {
                type: Object || String,
                required: true,
            },

            /**
             * Moment.js format string
             * @see https://momentjs.com/docs/#/displaying/
             */
            format: {
                type: String,
                default: 'lll',
            },
        },

        computed: {
            formattedDate() {
                return moment(this.date).format(this.format)
            },
        },
    }
</script>
