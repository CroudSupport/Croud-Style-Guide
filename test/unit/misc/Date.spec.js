import Vue from 'vue'
import moment from 'moment'
import Date from '../../../src/components/shared/misc/Date'

const Constructor = Vue.extend(Date)

const vm = new Constructor({
    propsData: {
        date: moment(),
    },
}).$mount()

describe('Date', () => {
    it('should match the snapshot', () => {
        expect(vm.$el).toMatchSnapshot()
    })
})
