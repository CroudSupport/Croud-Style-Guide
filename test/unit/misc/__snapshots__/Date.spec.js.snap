// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date should match the snapshot 1`] = `
<span>
  Aug 7, 2017 5:25 PM
</span>
`;
